tests/integ_tests/heartbeat_tests.rs

```diff
@@ -83,3 +84,80 @@ async fn activity_heartbeat() {
     );
     core.complete_execution(&task.run_id).await;
 }
+
+#[tokio::test]
+async fn many_act_fails_with_heartbeats() {
+    let (core, task_q) = init_core_and_create_wf("many_act_fails_with_heartbeats").await;
+    let activity_id = "act-1";
+    let task = core.poll_workflow_activation().await.unwrap();
+    // Complete workflow task and schedule activity
+    core.complete_workflow_activation(WorkflowActivationCompletion::from_cmd(
+        task.run_id,
+        ScheduleActivity {
+            seq: 0,
+            activity_id: activity_id.to_string(),
+            activity_type: "test_act".to_string(),
+            task_queue: task_q,
+            start_to_close_timeout: Some(Duration::from_secs(30).into()),
+            retry_policy: Some(RetryPolicy {
+                initial_interval: Some(Duration::from_millis(10).into()),
+                backoff_coefficient: 1.0,
+                maximum_attempts: 4,
+                ..Default::default()
+            }),
+            heartbeat_timeout: Some(Duration::from_secs(1).into()),
+            ..Default::default()
+        }
+        .into(),
+    ))
+    .await
+    .unwrap();
+
+    // Multiple times, poll for the activity, heartbeat, and then immediately fail
+    // Poll activity and verify that it's been scheduled with correct parameters
+    for i in 0u8..=3 {
+        let task = core.poll_activity_task().await.unwrap();
+        let hb_details = assert_matches!(task.variant, Some(activity_task::Variant::Start(s)) => s);
```

Comment:
The should be a better name than `hb_details` for this, it's just the activity start message..